Gems/ScriptCanvas/Code/Editor/View/Windows/Tools/UpgradeTool/Scanner.cpp

```diff
@@ -79,9 +79,10 @@ namespace ScriptCanvasEditor
             AZ::SystemTickBus::Handler::BusConnect();
         }
 
-        void Scanner::FilterAsset(SourceHandle asset)
+        void Scanner::FilterAsset(SourceHandle asset, bool isFromXMLSource)
         {
-            if (m_config.filter && m_config.filter(asset) == ScanConfiguration::Filter::Exclude)
+            if ((m_config.onlyIncludeLegacyXML && (!isFromXMLSource))
```

Comment:
a little confused, why onlyincludelegacyxml and isfromxmlsource need to be inverted to satisfy the case?

Reply:
Because it's looking for the file format. The actual filter operates on the graph itself, after it has been instantiated.